persistence-dse-6.8/src/main/java/io/stargate/db/dse/impl/DsePersistence.java

```diff
@@ -440,7 +440,10 @@ private DseConnection() {
     @SuppressWarnings("RxReturnValueIgnored")
     private DseConnection(@Nullable ClientInfo clientInfo, ClientState clientState) {
       super(clientInfo);
-      this.clientState = clientState;
+      this.clientState =
+          clientState instanceof StargateClientState
```

Comment:
how about doing this in the static `StargateClientState.from(ClientState cs)` and hide the constructor?